packages/custom_lint/test/cli_process_test.dart

```diff
@@ -72,7 +72,7 @@ void main() {
   test_app2/lib/main2.dart:1:6 • Hello world • hello_world
 ''',
     );
-    expect(process.exitCode, 1);
+    expect(process.exitCode, 0);
```

Comment:
That's not right. If a lint if emitted, it should emit 1 not 0.

Reply:
I'm considering the following configuration.
1. ErrorSeverity.info => Optional error (default exit code 0)
2. ErrorSeverity.warning => Default error, but optionally skippable (default exit code 1)
3. ErrorSeverity.error => Always error (default exit code 1)

This is consistent with the terminology used [in Dart analyze](https://github.com/dart-lang/sdk/blob/80bc65bb0b529b33023b4e812bd328663609921a/pkg/dartdev/lib/src/commands/analyze.dart#L44).

modifying the parameter definition [here](https://github.com/yamarkz/dart_custom_lint/blob/8fffd8d21664d4607fca402c88c9dd737d2fa476/packages/custom_lint/bin/custom_lint.dart#L12). So it will be changed to return an exit code of 0.

Could you please share your thoughts on the design for this ErrorSeverity? I would appreciate hearing your considerations and insights.

I have contemplated aligning it with the fundamental design of Dart analyzer. 
However, I am unsure if it aligns with the overall philosophy of this package.

Reply:
I would prefer basing custom_lint on `flutter analyze` rather than `dart analyze`.

With `flutter analyze`, infos lead to an exitCode 1 too, and you do `flutter analyze --no-fatal-infos`